File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/DeploymentEnvironmentDto.java
package de.sovity.authorityportal.api.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Connector deployment environment.")
public class DeploymentEnvironmentDto {
    @Schema(description = "Deployment environment ID", requiredMode = Schema.RequiredMode.REQUIRED)
    private String environmentId;
    @Schema(description = "Environment localized name", requiredMode = Schema.RequiredMode.REQUIRED)
    private String title;
    @Schema(description = "DAPS Token URL", requiredMode = Schema.RequiredMode.REQUIRED)
    private String dapsTokenUrl;
    @Schema(description = "DAPS JWKS URL", requiredMode = Schema.RequiredMode.REQUIRED)
    private String dapsJwksUrl;
    @Schema(description = "Logging House URL", requiredMode = Schema.RequiredMode.REQUIRED)
    private String loggingHouseUrl;
}

File: authority-portal-backend/authority-portal-api/src/main/java/de/sovity/authorityportal/api/model/DeploymentEnvironmentDto.kt
/*
 * Copyright (c) 2024 sovity GmbH
 *
 * This program and the accompanying materials are made available under the
 * terms of the Apache License, Version 2.0 which is available at
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contributors:
 *      sovity GmbH - initial implementation
 */

package de.sovity.authorityportal.api.model

import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Connector deployment environment.")
data class DeploymentEnvironmentDto(
    @field:Schema(description = "Deployment environment ID", requiredMode = Schema.RequiredMode.REQUIRED)
    val environmentId: String,
    @field:Schema(description = "Environment localized name", requiredMode = Schema.RequiredMode.REQUIRED)
    val title: String,
    @field:Schema(description = "DAPS Token URL", requiredMode = Schema.RequiredMode.REQUIRED)
    val dapsTokenUrl: String,
    @field:Schema(description = "DAPS JWKS URL", requiredMode = Schema.RequiredMode.REQUIRED)
    val dapsJwksUrl: String,
    @field:Schema(description = "Logging House URL", requiredMode = Schema.RequiredMode.REQUIRED)
    val loggingHouseUrl: String,
)
